WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -677,6 +693,61 @@ private extension EditableOrderViewModel {
         analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
     }
 
+    /// Adds a selected product variation (from the product variation list) to the order.
+    ///
+    func addProductVariationToOrder(_ variation: ProductVariation) {
+        let input = OrderSyncProductInput(product: .variation(variation), quantity: 1)
+        orderSynchronizer.setProduct.send(input)
+
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
+    }
+
+    /// Removes a selected product (from the product list) to the order.
+    ///
+    func removeProductFromOrder(_ product: Product) {
+        guard let orderItem = orderSynchronizer.order.items.first(where: { $0.productID == product.productID }) else {
+            DDLogError("Unable to find product ID: \(product.productID) in Order")
+            return
+        }
+        removeItemFromOrder(orderItem)
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductRemove(flow: flow.analyticsFlow))
+    }
+
+    /// Removes a selected product variation (from the product variation list) to the order.
+    ///
+    func removeProductVariationFromOrder(_ variation: ProductVariation) {
+        guard let orderItem = orderSynchronizer.order.items.first(where: { $0.variationID == variation.productVariationID }) else {
+            DDLogError("Unable to find product variation ID: \(variation.productVariationID) (parent ID: \(variation.productID)) in Order")
+            return
+        }
+        removeItemFromOrder(orderItem)
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductRemove(flow: flow.analyticsFlow))
+    }
+
+    /// Handles a product by adding it to the Order if selected, or removing it from the Order if unselected
+    ///
+    func handleProduct(_ product: Product) {
+        if !selectedProducts.contains(product) {
+            selectedProducts.append(product)
+            addProductToOrder(product)
+        } else {
+            selectedProducts.removeAll(where: { $0?.productID == product.productID })
+            removeProductFromOrder(product)
+        }
+    }
+
+    /// Handles a product variation by adding it to the Order if selected, or removing it from the Order if unselected
+    ///
+    func handleProductVariation(_ variation: ProductVariation, parent product: Product) {
```

Comment:
Same here about function naming.

Reply:
Renamed to `addOrRemoveProductVariationFromOrder` here https://github.com/woocommerce/woocommerce-ios/commit/cb86a9852a0da3321f719b4a40662a1577e60a8d